### DISCUSSION TITLE
Use Mantine components inside NPM package

### DISCUSSION THREAD

**Discussion by bankangle** [mantinedev/mantine#6660] [Q&A] [answered]
hi everyone!

Is there any way to use Mantine components with custom NPM packages? I Got some views that I want to share between mustiple projects. And if I just put some `.tsx` that utilize some of Mantine components, I got `[@mantine/core] MantineProvider was not found in tree` on runtime. That happen even if I wrap my components with MantimeProvider inside NPM package. Of cource problem does not appear If I copy component from NPM package and paste inside targtet project

Is nesting Mantine components inside NPM packages even possible? Is there any manuals/examples of doing that?

Not sure if thats important, but I use React+TS with Vite on target apps. Here is NPM package's `package.json`:

```
{
  "name": "@priveta/shared",
  "version": "0.0.9",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "clean": "rimraf ./dist/",
    "copy-files": "copyfiles -u 1 ./src/tailwind.css ./dist/",
    "build": "npm run clean && npm run copy-files && npx tsc"
  },
  "files": [
    "./dist"
  ],
  "devDependencies": {
    "@types/react": "^18.3.3",
    "copyfiles": "^2.4.1",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "rimraf": "^6.0.1",
    "typescript": "^5.5.4"
  },
  "dependencies": {
    "@mantine/core": "^7.12.1",
    "@tabler/icons-react": "^3.12.0",
    "@types/js-cookie": "^3.0.6",
    "js-cookie": "^3.0.5"
  }
}

```

and `tsconfig.json`:

```
{
  "compilerOptions": {
    "lib": [
      "DOM",
      "ESNext"
    ],
    "target": "ESNext",
    "module": "ESNext",
    "moduleResolution": "Node",
    "outDir": "./dist",
    "declaration": true,
    "noImplicitAny": true,
    "strict": true,
    "allowSyntheticDefaultImports": true,
    "jsx": "react"
  },
  "include": [
    "src"
  ],
  "exclude": [
    "node_modules",
    "dist"
  ]
}
```

And my target app sees my components, runs and builds the whole app, but error appears whan TSX component from package rendered

**Comment by rtivital** [comment, marked as answer]
Your project includes multiple instances of `@mantine/core` package – one included by the library and one installed by the app itself. Make sure that you have only one `@mantine/core` package instance in your application.

**Reply by bankangle** [reply]
confirm - moved ` "@mantine/core"` from `"dependencies"` to `"devDependencies"` in my package - and it works
wrapping shared component in my package with `<MantineProvider>` was a sily idea. I should better learn how packages works
thanks for quick response and beautiful UI lib!
